File: DictionaryProject/Commands/LeaveCommand.vb
<Highlight(RegEx:="^\s*leave ", Color:="Green")> _
Public Class LeaveCommand
    Inherits CommandBase

    Private _Amount As Integer
    Public Property Amount() As Integer
        Get
            Return _Amount
        End Get
        Set(ByVal value As Integer)
            _Amount = value
        End Set
    End Property

    Public Overrides Sub CheckSanity(ByVal project As Project)
        If _Amount < 0 Then
            project.Warnings.Add(String.Format("Line {0}: Leave command requires a positive integer or zero as its second argument.", m_lineNumber))
        End If
    End Sub

    Public Overrides Sub Execute(ByVal context As Context)
        If context.Tokens.Count > _Amount Then
            context.Tokens.RemoveRange(_Amount, context.Tokens.Count - _Amount)
        End If
    End Sub

    Public Overrides Sub LoadCommand(ByVal project As Project, ByVal reader As System.IO.TextReader, ByVal line As String, ByRef lineNumber As Integer)
        MyBase.LoadCommand(project, reader, line, lineNumber)

        Dim parts As List(Of String) = ProjectSerializer.ReadTokens(line)

        If parts.Count = 1 Then
            _Amount = 1
        ElseIf parts.Count = 2 Then
            If Not Integer.TryParse(parts(1), _Amount) Then
                project.Warnings.Add(String.Format("Line {0}: Leave command requires a positive integer or zero as its second argument.", m_lineNumber))
            End If
        End If
    End Sub

    Public Overrides Sub WriteCommand(ByVal writer As System.IO.TextWriter)
        writer.WriteLine("Leave {0}", _Amount)
    End Sub
End Class

File: monowordbuilder/wordbuilderbase/Commands/LeaveCommand.cs
using System.Collections.Generic;

namespace Whee.WordBuilder.Model.Commands
{
	[Highlight(RegEx = "^\\s*leave ", Color = "Green")]
	public class LeaveCommand : CommandBase
	{
	
		private int _Amount;
		public int Amount {
			get { return _Amount; }
			set { _Amount = value; }
		}
	
		public override void Execute(Context context)
		{
			if (context.Tokens.Count > _Amount) {
				context.Tokens.RemoveRange(_Amount, context.Tokens.Count - _Amount);
			}
		}

        public override void LoadCommand(Whee.WordBuilder.ProjectV2.IProjectSerializer serializer)
        {
            m_lineNumber = serializer.LineNumber;

            bool found = false;
            double amount = 0.0;
            ProjectV2.Token amountToken = serializer.ReadNumericToken(this, ref amount, out found);

            if (amountToken != null)
            {
                _Amount = (int)amount;
                if (serializer.ReadTextToken(this) != null)
                {
                    serializer.Warn("The leave command requires zero or one argument.", this);
                }
            }
            else if (found)
            {
                serializer.Warn("The leave command requires the first argument to be a positive integer.", this);
            }
            else
            {
                _Amount = 1;
            }
        }
	
		public override void WriteCommand(System.IO.TextWriter writer)
		{
			writer.WriteLine("Leave {0}", _Amount);
		}

        public override void CheckSanity(Project project, Whee.WordBuilder.ProjectV2.IProjectSerializer serializer)
        {
            if (_Amount < 0)
            {
                serializer.Warn("Leave command requires a positive integer or zero as its second argument.", this);
            }
        }
    }
}